torch/csrc/jit/operator_upgraders/upgraders.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <c10/macros/Export.h>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+namespace torch {
+namespace jit {
+
+class UpgradersMap {
+ public:
+  void set_content(const std::unordered_map<std::string, std::string>& content);
+  int count();
+  std::unordered_map<std::string, std::string> get_content();
+
+ private:
+  std::unordered_map<std::string, std::string> content_;
+  std::mutex lock;
```

Comment:
`shared_mutex` is more appropriate I think:
https://en.cppreference.com/w/cpp/thread/shared_mutex